### PR TITLE
Stability: Fix Gateway Kube2e Tests For Nightlies

### PR DESCRIPTION
**Pull request by sam-heilbron**
# Description

Ensure that Gateway Kube2e test suite will succeed when running nightly tests

# Context
Nightly tests are failing and we need to resolve them: [slack context](https://solo-io-corp.slack.com/archives/C04CJMXAH7A/p1680672330708599)

# Testing

## Nightly Tests
I trigged our nightly test job using this branch. The results can be found: https://github.com/solo-io/gloo/actions/runs/4629144600

## Regression Tests
Run the regression tests in the Gateway suite (specifically the ones that validate the webhook). When setting up the kind cluster, ensure that you specify the version of kubernetes that we [supply in our nightly runs](https://github.com/solo-io/gloo/blob/master/.github/workflows/nightly-tests.yaml#L28):

```
CLUSTER_NODE_VERSION=v1.23.13@sha256:ef453bb7c79f0e3caba88d2067d4196f427794086a7d0df8df4f019d5e336b6 VERSION=1.0.0-local KUBE2E_TEST=gateway  ./ci/deploy-to-kind-cluster.sh
```

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: test/kube2e/gateway/gateway_test.go [hunk 1]
```diff
@@ -2008,7 +2011,13 @@ spec:
   virtualHoost: {}
 `,
 							// This is handled by validation schemas now
-							expectedErr: `Error from server (BadRequest): error when creating "STDIN": VirtualService in version "v1" cannot be handled as a VirtualService: strict decoding error: unknown field "spec.virtualHoost"`,
+							// We support matching on number of options, in order to support our nightly tests,
+							// which are run using our earliest and latest supported versions of Kubernetes
+							errorMatcher: Or(
+								// This is the error returned when running Kubernetes <1.25
```
Comment (nfuden) [inline test/kube2e/gateway/gateway_test.go:2017]: 🥳

**Comment by nfuden** [comment]
pre merge lets run nightlies off this branch and link it!

File: test/kube2e/gateway/gateway_test.go (hunk 1 above)
Comment (jenshu) [reply test/kube2e/gateway/gateway_test.go:2017]: ahhh oops

File: changelog/v1.14.0-beta20/gateway-kube2e-nighlies.yaml
Comment (nfuden) [inline changelog/v1.14.0-beta20/gateway-kube2e-nighlies.yaml:1]: Can you update this to rc1?

**Review by jenshu — APPROVED** [review]

**Review by nfuden — APPROVED** [review]

**Pull request merged**